### PR TITLE
CBeer lasagna-master

### PR DESCRIPTION
**Pull request by beerparty**

### CODE REVIEW

File: lasagna-master/lasagna_master.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package lasagna
+
+// PreparationTime returns the estimate for the total preparation time
+// as an int based on the number of layers.
+func PreparationTime(layers []string, prepTimePerLayer int) int {
+  if prepTimePerLayer < 1 {
+    prepTimePerLayer = 2
+  }
+  prepTime := prepTimePerLayer * len(layers)
```
Comment (davidmogar) [inline lasagna-master/lasagna_master.go:9]: There's no need to declare a variable just to return it when it's so sort

File: lasagna-master/lasagna_master.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  return prepTime
+}
+
+// Quantities determines the quantity of noodles and sauce needed to make
+// your meal. The result should be returned as two values of noodles as
+// an int and sauce as a float64.
+func Quantities(layers []string) (int, float64) {
+  var qtyNoodles int = 0
+  var qtySauce float64 = 0.0
+  for i:=0; i<len(layers); i++ {
```
Comment (davidmogar) [inline lasagna-master/lasagna_master.go:19]:
```suggestion
  for _, layer := range layers {
```

File: lasagna-master/lasagna_master.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// your friend's list to the end of your list.
+func AddSecretIngredient(friendsList, myList []string) []string {
+  newList := make([]string, len(myList))
+  copy(newList, myList)
+  if len(friendsList) > 0 {
+    secretIngredient := friendsList[len(friendsList)-1]
+    newList = append(newList, secretIngredient)
+  }
+  return newList
+}
```
Comment (davidmogar) [inline lasagna-master/lasagna_master.go:39]: There's no need to make a copy as append creates a new slice. You can replace everything with:
```
return append(myRecipe, friendList[len(friendRecipe)-1])
```

File: lasagna-master/lasagna_master.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+// ScaleRecipe calculates the amounts for different numbers of portions.
+func ScaleRecipe(amounts []float64, portions int) []float64 {
+  var scalePortion float64 = float64(portions) / 2.0
+  newAmounts := make([]float64, len(amounts))
+  for i:=0; i<len(amounts); i++ {
```
Comment (davidmogar) [inline lasagna-master/lasagna_master.go:45]:
```suggestion
  for _, amount := range amounts {
```

**Review by davidmogar — COMMENTED** [review]
Could you run the formatter?

File: lasagna-master/lasagna_master.go (hunk 4 above)
Comment (beerparty) [reply lasagna-master/lasagna_master.go]: fixed

File: lasagna-master/lasagna_master.go (hunk 2 above)
Comment (beerparty) [reply lasagna-master/lasagna_master.go]: fixed

File: lasagna-master/lasagna_master.go (hunk 3 above)
Comment (davidmogar) [reply lasagna-master/lasagna_master.go:37]: This is still unresolved and it has a big impact as it is right now. The rest of my comment goes towards explaining slices and how functions work over them. If you understand this and you just wanted to acknowledge my comment above by resolving it, you can ignore the rest of this comment.

You should think on slices like an structure that holds a pointer to an array of data. This structure grows automatically whenever you add items and there's no more capacity on it.
There are several functions that can operate over slices, but not all of them behave in the same way. For example, `bytes.Split` takes a slice and a separator, splits the slice by the separator, and returns a slice of byte slices. The elements of the new slices still point to the original data.

Let's see an example!

```go
package main

import (
	"bytes"
	"fmt"
)

func main() {
	slice := []byte{'h', 'e', 'l', 'l', 'o'}
	newSlice := bytes.Split(slice, []byte("ll"))

	fmt.Printf("Original: %d\n", &slice[4])
	fmt.Printf("New: %d\n", &newSlice[1][0])
}
```

In here I create a slice of bytes and I split it by the string `ll`. Then I print the address of the last element of the original slice and the address of the element in the new slice that represents the `o` in the original slice. The result is the following:
```
Original: 824634466324
New: 824634466324
```

As you can see, both of them point to the same address.

Another example is `append` which returns a new slice containing the elements of an existing slice and a new element passed to the function. Now, the underlying array to which the original slice points to, won't be modified. Instead, the data will be copied to a new space in the memory.
In a similar way, `copy` will copy the content of a slice into another one. Both slices will not share the same array.

Here's an example:
```go
package main

import "fmt"

func main() {
	slice := []int{1, 2, 3}
	newSlice := append(slice, 4)

	sliceToCopyTo := make([]int, len(slice))
	copy(sliceToCopyTo, slice)

	fmt.Printf("Original: %d\n", &slice[0])
	fmt.Printf("Append: %d\n", &newSlice[0])
	fmt.Printf("Copy: %d\n", &sliceToCopyTo[0])
}
```
I'm using `append` and `copy` and in both cases I'm passing the same slice as source. Then I print the address of the first element for each of them, which will output something like the following:
```
Original: 824633819664
Append: 824633827904
Copy: 824633819688
```
As you see, all of them point to different memory addresses.

Going back to your code, you are replicating the original data twice. The first time you do it explicitly by calling `copy`. The second time you do it as part of the call to `append`. The ideal code for this function would be the one below, as you only copy the underlying array once, getting the data you want and leaving the original slice intact.
```go
return append(myRecipe, friendList[len(friendRecipe)-1])
```

Comment (beerparty) [reply lasagna-master/lasagna_master.go]: I didn't mean to ignore your comment, I forgot to push the change that had the fix in it. Thanks for the detailed explanation though.

**Pull request merged**